tests/test_shipment.py

```diff
@@ -1,8 +1,73 @@
 import unittest
 import gitlab
-from unittest.mock import MagicMock, patch
 import os
-from oar.core.shipment import GitLabMergeRequest, ShipmentData
+from unittest.mock import MagicMock, patch
+from oar.core.shipment import GitLabMergeRequest, ShipmentData, GitLabServer
+from oar.core.exceptions import (
+    GitLabServerException,
+    ShipmentDataException
+)
+
+class TestGitLabServer(unittest.TestCase):
+    @patch('os.getenv', return_value="test-token")
+    @patch('gitlab.Gitlab')
+    def setUp(self, mock_gitlab, mock_getenv):
+        self.mock_gl = MagicMock()
+        mock_gitlab.return_value = self.mock_gl
+        self.server = GitLabServer("https://gitlab.cee.redhat.com", "test-token")
+
+    def test_get_username_by_email_success(self):
+        # Setup mock user response
+        mock_user = MagicMock()
+        mock_user.email = "test@example.com"
+        mock_user.username = "testuser"
+        self.mock_gl.users.list.return_value = [mock_user]
+
+        # Test valid email
+        username = self.server.get_username_by_email("test@example.com")
+        self.assertEqual(username, "testuser")
+        self.mock_gl.users.list.assert_called_once_with(search="test@example.com")
+
+    def test_get_username_by_email_not_found(self):
+        # Setup empty response
+        self.mock_gl.users.list.return_value = []
+
+        # Test email not found
+        username = self.server.get_username_by_email("notfound@example.com")
+        self.assertIsNone(username)
+
+    def test_get_username_by_email_invalid(self):
+        # Test invalid email formats
+        with self.assertRaises(ValueError):
+            self.server.get_username_by_email("")
+        
+        with self.assertRaises(ValueError):
+            self.server.get_username_by_email("invalid-email")
+
+    def test_get_username_by_email_api_error(self):
+        # Setup API error
+        self.mock_gl.users.list.side_effect = gitlab.exceptions.GitlabError("API error")
+
+        with self.assertRaises(GitLabServerException) as context:
+            self.server.get_username_by_email("test@example.com")
+        self.assertIn("GitLab API error", str(context.exception))
+
+    def test_get_username_by_email_real(self):
+        """Test with real GitLab API (requires GITLAB_TOKEN env var)"""
+        if not os.getenv('GITLAB_TOKEN'):
+            self.skipTest("GITLAB_TOKEN not set - skipping real API test")
+            
+        try:
+            # Use a known Red Hat email that should exist in GitLab
+            test_email = "rioliu@redhat.com"
+            server = GitLabServer("https://gitlab.cee.redhat.com")
+            username = server.get_username_by_email(test_email)
+            self.assertIsNotNone(username)
+            self.assertIsInstance(username, str)
+            print(f"\nFound GitLab username for {test_email}: {username}")
```

Comment:
Should't this be an assert?

Reply:
I don't want to put my userid in the test case :)